google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/GoogleComputeEngineService.java

```diff
@@ -125,37 +124,30 @@
    protected synchronized void cleanUpIncidentalResourcesOfDeadNodes(Set<? extends NodeMetadata> deadNodes) {
       Set<String> orphanedGroups = findOrphanedGroups.apply(deadNodes);
       for (String orphanedGroup : orphanedGroups) {
-         cleanUpNetworksAndFirewallsForGroup(orphanedGroup);
+         cleanUpFirewallsForGroup(orphanedGroup);
       }
    }
 
-   private void cleanUpNetworksAndFirewallsForGroup(final String groupName) {
-      String resourceName = namingConvention.create().sharedNameForGroup(groupName);
-      Network network = api.networks().get(resourceName);
+   private void cleanUpFirewallsForGroup(final String groupName) {
+      GroupNamingConvention namingScheme = namingConvention.create();
       FirewallApi firewallApi = api.firewalls();
 
       for (Firewall firewall : concat(firewallApi.list())) {
-         if (firewall == null || !firewall.network().equals(network.selfLink())) {
+         if (firewall == null) {
```

Comment:
Can this be true at some point? Does the API return `null` objects int he returned Iterable?